src/components/data.jsx

```diff
@@ -1,177 +1,210 @@
+      setSessionStorage = () => {
+        sessionStorage.setItem('language', filters.language);
+        sessionStorage.setItem('stars', filters.stars);
+        sessionStorage.setItem('keyword', filters.keyword);
+      };
+
+      setSessionStorage();
+
+      signal.addEventListener('abort', () => {
+        console.log(`fetch request for ${filters.url} aborted!`);
+      });
     }
-    
-    setSessionStorage()
-        
-      
-  },[filters.url]); 
-  
-
-
-  function displayresults(){
-    let displaytext=''
-
-      if(filters.loading){
-        displaytext = 'Loading Data Please Wait...'
-      }else{
-        if(filters.count>0){
-          displaytext =  `Displaying ${filters.count} results.`
-        } else {
-          displaytext = 'Search Filters returned no results. Try changing search filters.'
-        }
+    //clean up after unmount
+    return () => {
+      controller.abort();
+      setData(null);
+      setSessionStorage();
+      resetSubmit();
+    };
+  }, [filters.url]);
+
+  function displayresults() {
+    let displaytext = '';
+
+    if (filters.loading) {
+      displaytext = 'Loading Data Please Wait...';
+    } else {
+      if (filters.count > 1) {
+        displaytext = `Displaying ${filters.count} results.`;
+      } else if (filters.count === 1) {
+        displaytext = `Displaying ${filters.count} result.`;
+      } else {
+        displaytext =
+          'Search Filters returned no results. Try changing search filters.';
       }
-      return displaytext
+    }
+    return displaytext;
   }
 
-  //console.log(readMore)
-  function textMin(text,keyNo){
-      return(
-        <TextMinimize keyNo={keyNo} text={text} setEmpty={setReadMoreEmpty}/>
-      )
+  function textMin(text, keyNo) {
+    return (
+      <TextMinimize keyNo={keyNo} text={text} setEmpty={setReadMoreEmpty} />
+    );
   }
 
   const ItemIncrement = (function () {
-    let counter = 0
-    return function () {counter += 1; return counter}
-  })();//closure with self invoked function:)
-
-
+    let counter = 0;
+    return function () {
+      counter += 1;
+      return counter;
+    };
+  })();
 
   return (
-    
-      <div>
-        
-        <p>{displayresults()}</p>
-        <p>{error?`Error: ${error}`:null}</p>
-
-          <ul>
-            
-            {data?data.map((item,index) =>
-            <>
-            <div id='repo-items'>
-              <div id='repo-list-items'>
-                <div id='order-number'>
-                {ItemIncrement()}
+    <div>
+      <p></p>
+      <p>{error ? `Error: ${error}` : displayresults()}</p>
+
+      <ul>
+        {data
+          ? data.map((item, index) => (
+              <>
+                <div id="repo-items">
+                  <div id="repo-list-items">
+                    <div id="order-number">{ItemIncrement()}</div>
+
+                    <li
+                      id="repo-list-items-name"
+                      key={`repo-list-items-name-${index}`}
```

Comment:
hmm I think there is a misunderstanding about keys. Basically you don't need to use `key` here. Check here for more information -> https://reactjs.org/docs/lists-and-keys.html

Reply:
keys are only needed if there is a list (arrays, etc...)